### ISSUE TITLE
Make events shareable modal and add to public event card

### ISSUE THREAD

**Issue by bakeiro** [Couchers-org/couchers#4878]
Blocked by #5008  , #4981  and #4980 

**Current situation:**
I see a fantastic and cool event in couchers (as always :D) I want to share this event with my friends & family so I need a button/shortcut or something so I can share it

**Desired situation:**
Implement a share button in the event list which opens a share modal

ToDo:
- Implement a share button [this goes on the event list item component, not the event page itself see `app/web/features/communities/events/EventCard.tsx`]
![image](https://github.com/user-attachments/assets/cce4c64d-ea26-44a3-8156-cf8ca6709c04)
- Implement another share button in the event's page `app/web/features/communities/events/EventPage.tsx` [image for inspiration]
![Image](https://github.com/user-attachments/assets/276e7aa0-f4b5-4d1a-b31b-ddb57a85a9a9)
- Implement this modal to share the choosed event
![370201604-94304fb0-abd8-4c88-aae5-5b11b262840c](https://github.com/user-attachments/assets/9f2c94ae-0371-4b8e-8cd6-b9bc10175c67)
- Code each button action
- Keep responsive design in mind, should work fine in desktop + mobile

**Comment by bakeiro** [comment]
<details><summary>figma notes, unrelevant for the dev:</summary>
<p>
Official Figma designs to share profiles
<img src="https://github.com/user-attachments/assets/94304fb0-abd8-4c88-aae5-5b11b262840c" />
</p>
</details>

**Comment by jesseallhands** [comment]
Can we do the same thing as in this design, but simply have a preview of the publicly viewable snippet at the top?

![370201604-94304fb0-abd8-4c88-aae5-5b11b262840c](https://github.com/user-attachments/assets/f70f349f-6b1d-488e-9ac4-f34ceabe1b92)

Keep in mind the icons pictured are just representative placeholders - the final design doesn't need to have icons that look just like that. They should all be consistent in style though.

**Comment by bakeiro** [comment]
> Can we do the same thing as in this design, but simply have a preview of the publicly viewable snippet at the top?

I kind of get the idea, but I guess would be better for the dev to know exactly how should it look, so they can know what to implement/code

**Comment by aapeliv** [comment]
Let’s move the share screen to another issue.

**Comment by bakeiro** [comment]
> Let’s move the share screen to another issue.

@aapeliv, yeah... the thing is that all share-event tickets are created... and I don't think makes sense to split this one if we do, is to create just the share button without any functionality, so, not sure if makes sense to move this to another ticket...

**Comment by nabramow** [comment]
@aapeliv @bakeiro This is blocked by #5008 , #4981 and #4980 is that right? Seems they need to be done first?

**Comment by bakeiro** [comment]
@nabramow yes and no, we can start creating the components, the features and don't merge it until the other tickets are ready
